book/src/exec.md

```diff
@@ -0,0 +1,168 @@
+
+We call a **partial transaction** a state transition where the first requirement is satisfied (all VPs evaluate to True), but the second one isn't (the state transition is unbalanced). Such a state transition isn't a valid transaction and cannot be published on the blockchain, 
+but can be combined with other partial transactions in order to build a valid transaction. 
+
+![img.png](img/exec_partial_tx.png)
+
+Partial transactions are immutable objects, and once they are created, they cannot be changed. 
+In principle, any party can create a partial transaction, if they can prove that the VPs of the parties involved in the partial transaction are satisfied. 
+
+**Initial partial transactions** are the partial transactions that users create at the beginning, spending the notes that will be given to the counterparty once the intents are matched.
+
+**Note**: For simplicity we assume that intents imply both sending notes and receiving notes.
+
+As a part of the initial partial transaction, users also output **intent token notes**. 
+### Intent tokens
+
+**Intent tokens** are a mechanism that makes sure that the partial transaction is *unbalanced* until the corresponding intent is satisfied, making it impossible to publish the transaction.
+
+Intent token notes are **dummy** notes - meaning that unlike "normal" notes, the merkle path isn't checked for them (but they can have arbitrary value and stored in the CMtree, just like "normal"). 
+
+How intent tokens are used:
+1. Spending notes in the initial partial transaction, the user additionally outputs intent token note of value [1].
+This note will only be spent if the user's intent (intentVP) is satisfied.
+2. Once all of the parties involved in a state transition spent their intent token notes (meaning that the intents of all of them are satisfied), the transaction is balanced and can be finalized.
+
+![img.png](img/exec_intent_notes.png)
+
+
+If we look at the intent token notes as messages passing from one user (application) to another, the output [1] intent note can signal a message being sent, and spending of that note will signal that the message has been received. See more about the idea of message passing [here](message_passing.md).
+
+## Step 2: Gossip
+Once the intentVP is specified and the initial partial transaction is created, the user sends everything to the intent gossip network. Gossip nodes send those intents to each other, spreading them around. Solvers listen to the intent gossip network and receive the intents.
+
+## Step 3: Solve
+
+Solvers receive intents from the intent gossip network and match them together in order to create transactions. 
+
+**Note**: that implies that users need to give some information to the solvers (including the content of `intentVP`) that is sufficient to create partial transactions and transactions (to create proofs).
```

Comment:
We should be more specific here - in order to match two partial transactions, what does the solver need?
- intent VPs of both partial transactions
- note commitments / nullifiers from prior partial transactions which will be chained
- proofs from prior partial transactions which will be chained

anything else?